bazel/repositories.bzl

```diff
@@ -209,218 +222,155 @@ def envoy_dependencies(path = "@envoy_deps//", skip_com_google_protobuf = False,
+        name = "com_github_bombela_backward",
+        build_file = "@envoy//bazel/external:backward.BUILD",
+    )
+    native.bind(
+        name = "backward",
+        actual = "@com_github_bombela_backward//:backward",
+    )
 
-def com_github_fmtlib_fmt(repository = ""):
-  native.new_http_archive(
-      name = "com_github_fmtlib_fmt",
-      urls = [
-          "https://github.com/fmtlib/fmt/releases/download/4.0.0/fmt-4.0.0.zip",
-      ],
-      sha256 = "10a9f184d4d66f135093a08396d3b0a0ebe8d97b79f8b3ddb8559f75fe4fcbc3",
-      strip_prefix = "fmt-4.0.0",
-      build_file = repository + "//bazel/external:fmtlib.BUILD",
-  )
-  native.bind(
-      name="fmtlib",
-      actual="@com_github_fmtlib_fmt//:fmtlib",
-  )
+def _com_github_cyan4973_xxhash():
+    _repository_impl(
+        name = "com_github_cyan4973_xxhash",
+        build_file = "@envoy//bazel/external:xxhash.BUILD",
+    )
+    native.bind(
+        name = "xxhash",
+        actual = "@com_github_cyan4973_xxhash//:xxhash",
+    )
+
+def _com_github_eile_tclap():
+    _repository_impl(
+        name = "com_github_eile_tclap",
+        build_file = "@envoy//bazel/external:tclap.BUILD",
+    )
+    native.bind(
+        name = "tclap",
+        actual = "@com_github_eile_tclap//:tclap",
+    )
 
-def com_github_gabime_spdlog(repository = ""):
-  native.new_http_archive(
-      name = "com_github_gabime_spdlog",
-      urls = [
-          "https://github.com/gabime/spdlog/archive/v0.14.0.tar.gz",
-      ],
-      sha256 = "eb5beb4e53f4bfff5b32eb4db8588484bdc15a17b90eeefef3a9fc74fec1d83d",
-      strip_prefix = "spdlog-0.14.0",
-      build_file = repository + "//bazel/external:spdlog.BUILD",
-  )
-  native.bind(
-      name="spdlog",
-      actual="@com_github_gabime_spdlog//:spdlog",
-  )
+def _com_github_fmtlib_fmt():
+    _repository_impl(
+        name = "com_github_fmtlib_fmt",
+        build_file = "@envoy//bazel/external:fmtlib.BUILD",
+    )
+    native.bind(
+        name = "fmtlib",
+        actual = "@com_github_fmtlib_fmt//:fmtlib",
+    )
+
+def _com_github_gabime_spdlog():
+    _repository_impl(
+        name = "com_github_gabime_spdlog",
+        build_file = "@envoy//bazel/external:spdlog.BUILD",
+    )
+    native.bind(
+        name = "spdlog",
+        actual = "@com_github_gabime_spdlog//:spdlog",
+    )
 
-def com_github_gcovr_gcovr(repository = ""):
-  native.new_git_repository(
-      name = "com_github_gcovr_gcovr",
-      remote = "https://github.com/gcovr/gcovr",
-      commit = "c0d77201039c7b119b18bc7fb991564c602dd75d",
-      build_file = repository + "//bazel/external:gcovr.BUILD",
-  )
-  native.bind(
-      name = "gcovr",
-      actual = "@com_github_gcovr_gcovr//:gcovr",
-  )
+def _com_github_gcovr_gcovr():
+    _repository_impl(
+        name = "com_github_gcovr_gcovr",
+        build_file = "@envoy//bazel/external:gcovr.BUILD",
+    )
+    native.bind(
+        name = "gcovr",
+        actual = "@com_github_gcovr_gcovr//:gcovr",
+    )
```

Comment:
Seems like there is a bunch of boilerplate that can be still cleaned up in the repetition above, but up to you, could be just done later.

Reply:
I want to get rid of these binds entirely (https://github.com/bazelbuild/bazel/issues/1952), replacing them with aliases somewhere under `@envoy//bazel/external`. Then loading Envoy into a workspace wouldn't pollute the namespace.

Reply:
If you can make it work fully with `alias`, that works with me. I can't remember exactly why I opted for `bind` in the first place.